modules/lambda_function/tf_module/api_gateway.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_apigatewayv2_api" "api" {
+  count         = var.expose_via_domain ? 1 : 0
+  name          = "opta-${var.module_name}-${random_string.lambda.result}"
+  protocol_type = "HTTP"
+}
+
+resource "aws_apigatewayv2_route" "route" {
+  count     = var.expose_via_domain ? 1 : 0
+  api_id    = aws_apigatewayv2_api.api[0].id
+  route_key = "ANY /"
+
+  target = "integrations/${aws_apigatewayv2_integration.lambda_integration[0].id}"
+}
+
+resource "aws_apigatewayv2_integration" "lambda_integration" {
+  count                  = var.expose_via_domain ? 1 : 0
+  api_id                 = aws_apigatewayv2_api.api[0].id
+  integration_type       = "AWS_PROXY"
+  payload_format_version = "2.0"
+  connection_type        = "INTERNET"
+  description            = "Lambda integration"
+  integration_method     = "POST"
```

Comment:
Are we supporting only POST requests?

Reply:
We're supporting any request method, but the integration with lamba MUST be post-- the data of the original method sent is passed along as part of the event data to the lambda function, as per the documentation